dotnet/src/SemanticKernel/Planning/IPlan.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.Planning;
+
+public interface IPlan
+{
+    /// <summary>
+    /// Goal of the plan
+    /// </summary>
+    string Goal { get; }
+
+    /// <summary>
+    /// State of the plan
+    /// </summary>
+    ContextVariables State { get; }
+
+    /// <summary>
+    /// Steps of the plan
+    /// </summary>
+    PlanStep Steps { get; }
```

Comment:
I think you can merge this and Goal, the plural name is also confusing given that it's the root of a tree

Reply:
`step` is a reserved word. Suggestions? Root? PlanStep?

Reply:
Going with Root for now.